### PR TITLE
Deinterlacing support for VideoSourcePlugin

### PR DESCRIPTION
**Pull request by Tyrrr**
I added deinterlacing support which makes 1080i or other interlaced sources watchable. It's possible to choose from all deinterlacing methods vlc provides. Also, in the VideoSourcePluginWrapper project you have awesomium.lib as a dependency, which seems like a mistake cause it isn't used at all and wouldn't make much sense anyway.
Would be nice if you could pull the changes.

### CODE REVIEW

File: Plugin/VideoSource.cpp [hunk 1]
```diff
@@ -326,6 +326,12 @@ void VideoSource::UpdateSettings()
 
     libvlc_media_list_player_set_playback_mode(mediaListPlayer, config->isPlaylistLooping ? libvlc_playback_mode_loop : libvlc_playback_mode_default);
     
+    if(config->deinterlacing == TEXT("none")) {
+        libvlc_video_set_deinterlace(mediaPlayer,NULL);
+    }
+    else {
+        libvlc_video_set_deinterlace(mediaPlayer,config->deinterlacing.CreateUTF8String());
```
Comment (kc5nra) [inline Plugin/VideoSource.cpp]: I think you need to free this string

**Comment by kc5nra** [comment]
Thanks for the pr, I'll test and merge after the change

**Comment by Tyrrr** [comment]
I freed the string now and changed the string array to a std::array and did some other changes

**Comment by Tyrrr** [comment]
I also think *rate = 44100; in AudioOutputStreamHandler::AudioSetupCallback should be changed to  *rate = OBSGetSampleRateHz(); The crackling seems to come from OBS resampling. Practically all of my sources are 48kHz and I stream with 48kHz. Currently VLC resamples my source to 44.1k and OBS resamples it again to 48k introducing crackling. With the change VLC resamples to the OBS output rate and OBS doesn't resample it again.
The change removes the crackling for me with one minor problem still lingering. If I change the OBS sample rate from 44.1k to 48k and restart the stream but do not open the VideoSourceplugin options again, the audio source and therefore the sampling rate don't get changed and the crackling persists until I open the VideoSourceplugin options.
I can commit the change if you agree.

**Comment by kc5nra** [comment]
Yeah, it used to be that OBS always used 44100 and I guess it changed at some point while I wasn't paying attention.  But feel free to submit a PR since I don't really have any media to test with.

As far as I can tell though, restarting the stream calls libvlc_audio_set_callbacks which leads to AudioSetupCallback being called again.  This sets the rate according to OBSGetSampleRateHz() and should operate as one would expect.  Is this somehow not happening?

**Comment by Tyrrr** [comment]
It's even weirder. If I change the sampling settings in OBS and restart the stream there is crackling, regardless if I changed it to 44.1k or 48k. If I then restart OBS it's fine again even if I restart the stream. Maybe it's a bug somewhere in OBS. Or it could be related to the async problem I have with audio output to stream. I'll make a separate Issue about that and submit a PR for the sampling rate fix.

**Pull request merged**